admin/workspace-providers/deployment.md

```diff
@@ -0,0 +1,176 @@
+---
+title: Deployment
+description: Learn how to deploy a workspace provider.
+state: beta
+---
+
+This article walks you through the process of deploying a workspace provider to
+a [Kubernetes cluster](../../setup/kubernetes/index.md).
+
+[Workspace Providers](index.md) are logical groups of resources to which
+developers can deploy environments. Like the Coder deployment, workspace providers
+are deployed via a helm chart to the Kubernetes cluster where you'd like to
+provision new environments.
+
+## Dependencies
+
+Install the following dependencies if you haven't already:
+
+- [Coder CLI](../../cli/installation.md)
+- [helm](https://helm.sh/docs/intro/install/)
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+## Requirements
+
+1. Workspace providers **must have a hostname set** that is a subdomain of the
+   Coder deployment. For example, if the Coder deployment's hostname is
+   `coder.example.com`, the workspace provider's hostname must match the format
+   `*.coder.example.com`.
+1. The main Coder deployment and the workspace provider must be able to
+   communicate bi-directionally via their respective hostnames.
+1. The Kubernetes cluster address must be reachable from the Coder deployment.
+
+## Connecting to the Cluster
+
+To add a Kubernetes cluster as a workspace provider, you must first make sure
+that you're connected to the cluster you want to expand into. Run the following
+command:
+
+```bash
+kubectl config current-context
+```
+
+Confirm that your current kubectl context correct before continuing; otherwise,
+connect to the correct context.
+
+## Creating the Coder Namespace (Optional)
+
+We recommend running workspace providers in a separate
+[namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/);
+to do so, run:
+
+```bash
+kubectl create namespace coder
+```
+
+Next, change the kubectl context to point to your newly created namespace:
+
+```bash
+kubectl config set-context --current --namespace=coder
```

Comment:
we should just opt for supplying `--namespace` to helm instead

Reply:
cc @f0ssel , @mterhar

Reply:
I disagree, this is in line with how we say to install coder normally, and the namespace for WSP will probably not be always be coder and having it at every step means they have to edit every step themselves.

Reply:
The problem with using "current context" like this is that it becomes a kind of ambient parameter, kind of like global variables: it's something that affects various steps, but not obviously so. It also makes it very easy to accidentally deploy things into the wrong namespace (c.f. our own Kubernetes clusters, where people frequently deploy their personal stuff into the coder-ci or preview clusters, for example)

I also understand, though, that we don't want to have people modify every command they're copying. So as a longer-term approach, I would propose that we provide install scripts rather than ask admins to copy and paste commands out of a guide like this, though I acknowledge that's outside the scope of this current release.